addons/purchase_stock/models/res_partner.py

```diff
@@ -25,11 +25,11 @@ def _compute_on_time_rate(self):
         for line in order_lines:
             on_time, ordered = partner_dict.get(line.partner_id, (0, 0))
             ordered += line.product_uom_qty
-            on_time += sum(line.mapped('move_ids').filtered(lambda m: m.state == 'done' and m.date <= datetime.combine(m.purchase_line_id.date_planned, time.min)).mapped('quantity_done'))
```

Comment:
Maybe a better solution would be to define the receipt date 1h after order date by default.

Reply:
But this can't solve the problem that when receiving product on the scheduled date, it can still be considered 'late' because the 'time' part of the "datetime" is late.

Reply:
If then vendor said that he deliver you everyday at 11am and he delivers the good at 4pm then he is late and we should be able to define a correct on time percentage

Reply:
@amoyaux If we still want to consider the 'time' part, then I think it's not necessary to change it 1 hour behind the the order date. Because in that case, user will definitely set the time manually. Plus, Remy set the default to be 12:00 to reduce the confusion between the data on replenishment report and the purchase order, which may show different date after a time zone conversion. what do you think?